### ISSUE TITLE
ContentButton, Pressed Event, View not updating

### ISSUE THREAD

**Issue by sethom** [builttoroam/BuildIt#54]
I just encountered a strange issue using a ContentButton that I can't explain. I've got a view with several conventional buttons on the page and I've switched one of them out for a contentbutton. When any of the buttons is pressed I call MyMethod on a ViewModel. 

Here's how this looks using Caliburn.Micro actions to wire up the Pressed event to the ViewModel action:

  <bitctrls:ContentButton WidthRequest="55"  x:Name="A"
                    HeightRequest="{Binding ButtonHeight}" Grid.Row="0" Grid.Column="0" Margin="0" 
                    VerticalOptions="CenterAndExpand" HorizontalOptions="CenterAndExpand" cal:Message.Attach="[Event Pressed] = [Action MyMethod('A')]">
                    <StackLayout VerticalOptions="CenterAndExpand" HorizontalOptions="CenterAndExpand">
                        <Label Text="A" TextColor="Blue" VerticalOptions="Center" HorizontalOptions="Center" />
                    </StackLayout>
                </bitctrls:ContentButton>

And now here's the same thing with a regular button in the same view:
 <Button WidthRequest="55" HeightRequest="{Binding ButtonHeight}" Grid.Row="0" Grid.Column="1" Text="B" Margin="0" VerticalOptions="CenterAndExpand" HorizontalOptions="CenterAndExpand" BorderWidth=".5" BorderColor="Red"
                    cal:Message.Attach="[Event Clicked] = [Action MyMethod('B')]" />

The ViewModel has a property on it called "Note" and the view has a label which is bound to "Note" and displays either 'A' or 'B' depending on which button was pressed.

Whenever I click on the ContentButton this Note label is not updated to display the value of 'A'. However, when I press the conventional button, B, the binding is updated to display 'B'. 

I've verified that the press event is being raised and that the ViewModel method is being called in both cases. I've tested by replacing the Caliburn Micro action syntax and using a Pressed event handler which calls the method directly on the ViewModel with the same result.

This illustrates that it doesn't have anything to do with Caliburn.Micro's idiosyncratic "Action" syntax. It seems to have something to do with how the Pressed event is being raised. Replacing the ContentButton with a normal Button fixes the problem. The view just isn't updating on this pressed event for some reason.

Is there something different about the way the "Pressed" event is being raised that might account for why the Xamarin Forms view is not being updated? Let me know if I need to provide more code in order to reproduce the issue.

**Comment by sethom** [comment]
I forgot to include the xaml for the regular button:

 <Button Text="B" 
                    cal:Message.Attach="[Event Clicked] = [Action MyMethod('B')]" />

**Comment by nickrandolph** [comment]
Can you try adding “await Task. Delay(50);” in your event handler and let me know if that helps?

**Comment by nickrandolph** [comment]
Also, are you seeing this behaviour on all platforms?

**Comment by sethom** [comment]
Thanks for the reply, btw!

I gave it a try. Didn't work. I noticed something else though. After I push the contentbutton and the view model property is updated with 'A' (with the view failing to update with this value).. subsequent updates to the viewmodel by pushing the contentbutton or other buttons causes the view to update with the value of 'A' as it should have the first time. So it ends up working later just not right after the pressed event.

I tried putting the await Task.Delay(50) both in the pressed event and in the viewmodel method. I also tried this in the pressed event:

 await Task.Run(async () => 
            {
                await Task.Delay(1000);
                ViewModel.UpdateNoteProperty('A');

            });

to just go out on a limb. This didn't work either.

I don't have a build in place right now for anything other than iOS. I could try to set this up if it would help out.
